projects/payment-lib/src/lib/components/reports/reports.component.ts

```diff
@@ -186,7 +191,40 @@ downloadReport(){
           }
         })
 
-    } else {
+    } else if(selectedReportName === 'TELEPHONY_PAYMENTS') {
+
+            this.paymentViewService.downloadTelephonyPaymentsReport(selectedStartDate,selectedEndDate).subscribe(
+              response =>  {
+                this.errorMessage = this.errorHandlerService.getServerErrorMessage(false, false, '');
+
+                const result = {data: JSON.parse(response)['telephony_payments_report_list']};
+                let res= {data: this.applyDateFormat(result)};
+                //res.data= telephonyPaymentsRptDefault;
+                if(result['data'].length > 0) {
+                   for( var i=0; i< res['data'].length; i++) {
+                     if(res['data'][i]["Amount"] !== undefined) {
+                        res['data'][i]['Amount'] = this.convertToFloatValue(res['data'][i]['Amount']);
+                     }
+                  }
+                }else{
+                  res.data = telephonyPaymentsRptDefault;
+                }
+                this.isDownLoadButtondisabled = false;
+                this.xlFileService.exportAsExcelFile(res['data'], this.getFileName(this.reportsForm.get('selectedreport').value, selectedStartDate, selectedEndDate ));
+
+              },
+              (error: any) => {
+                this.isDownLoadButtondisabled = false;
+                const errorContent = error.replace(/[^a-zA-Z ]/g, '').trim();
+                const statusCode = error.replace(/[^a-zA-Z0-9 ]/g, '').trim().split(' ')[0];
+                if(statusCode === '404') {
```

Comment:
remember that even in TypeScripot "any" can be shapeless.  That is why this are any. 

Try to typeSafe the any before 

 const errorContent = error.replace(/[^a-zA-Z ]/g, '').trim();
                const statusCode = error.replace(/[^a-zA-Z0-9 ]/g, '').trim().split(' ')[0];